### DISCUSSION TITLE
Using Move action on fieldArray Radio input

### DISCUSSION THREAD

**Discussion by douglasrcjames** [react-hook-form/react-hook-form#11572] [Q&A] [answered]
I am trying to move values from one position to another with the built in function with fieldArrays but the radio field seems to not switch properly. Using my codesandbox below, when I create just two array entries, labeling the first with all "1" type values and the second with "2" values, then attempt to click the "move" button, the other fields swap as expected, but the radio field will only swap the second value and the first value is not seen, as is "null" it seems. What am I missing here?
![image](https://github.com/react-hook-form/react-hook-form/assets/22306305/80d0bba9-5913-4261-93c4-e651c782d364)

https://codesandbox.io/p/sandbox/react-hook-form-usefieldarray-forked-vmddlh?file=%2Fsrc%2Findex.tsx

**Comment by sujinleeme** [comment, marked as answer]
Currently, the radio input is not controlled.

<img src="https://github.com/react-hook-form/react-hook-form/assets/6011401/bc902110-be47-4737-92d5-d7a12fa6c3d3" width="80%" />


What [the document](https://react-hook-form.com/docs/usefieldarray) recommends in the `useFieldArray()`:
>  Each input name needs to be unique, if you need to build checkbox or radio with the same name then use it with useController or Controller.

Eventually, you may need to create the controlled radio / radio group component with `useController` or `Controller`. 


e.g) [codesandbox](https://codesandbox.io/p/sandbox/gallant-mclaren-5jcnh7?file=%2Fsrc%2Findex.tsx)

```ts
const RadioButtonGroup = ({
  control,
  index: number,
}: {
  control: UseForm["control"];
  index: number;
}) => {
  const { field } = useController({
    control,
    name: `test.${number}.radioEx`,
    rules: {
      required: true,
    },
  });
  const { value, name, ...methods } = field;

  return (
    <div>
      <p style={{ display: "inline", margin: "0" }}>r-op 1</p>
      <input
        type="radio"
        {...methods}
        value="r-op 1"
        checked={value === "r-op 1"}
      />
      <p style={{ display: "inline", margin: "0" }}>r-op 2</p>
      <input
        type="radio"
        {...methods}
        value="r-op 2"
        checked={value === "r-op 2"}
      />
    </div>
  );
};
```

**Reply by douglasrcjames** [reply]
Gotcha, okay that makes sense. Do you know if any other input types are not controlled? Or is just the radio input type?

**Reply by sujinleeme** [reply]
Inside the plain form object, it works well. RHF checks the input type and gets value by ref https://github.com/react-hook-form/react-hook-form/blob/ec8a46e1e2566aab42f64509efaaec1210202c3c/src/logic/getFieldValue.ts#L12  I'm not sure about the core development part, but it seems like the only thing that isn't working in the array is `<input type=radio {...register('name')}` because same names are registered in different refs.

**Comment by Moshyfawn** [comment]
From [the `useFieldArray` docs](https://react-hook-form.com/docs/usefieldarray#rules):

> Each input name needs to be unique, if you need to build checkbox or radio with the same name then use it with useController or Controller.

**Comment by douglasrcjames** [comment]
Thanks y'all, this has helped me understand this usage a bit more!
